index.d.ts

```diff
@@ -0,0 +1,39 @@
+// copied from Redux so that we don't have to add it as a dependency
+interface Action<T = any> {
+  type: T
+}
+
+export type Enhancement = object;
+
+export type EnhancerId = string | symbol;
+
+type ActionEnhancerById = {
+  readonly id: EnhancerId;
+};
+
+type ActionEnhancerByType = {
+  actionTypes: string[];
+};
+
+export type ActionEnhancer<S, E extends Enhancement, V = any> = (
+  | ActionEnhancerById
+  | ActionEnhancerByType) & {
+  mapState: (state: S, value: V) => E;
+};
+
+export type UnenhancedAction<T> = Action<T> & {
+  [index: string]: any;
+};
+
+export type EnhancedAction<
+  A extends Action,
+  E1 extends Enhancement,
+  E2 extends Enhancement = E1,
```

Comment:
This looks a bit goofy, but it lets us pass in multiple `Enhancements` without a penalty for passing in less. i.e., you can pass in anywhere between 1 and 9 enhancements. Technically you'll get an error if you try to do more than nine, but I'm betting that if you're enhancing something that much you've got other problems. We could extend the list if we wanted to allow for more, but nine already felt like overkill to me.

We can't say "accept `n` generics" in TypeScript, so this is a workaround for that.

Reply:
womp yea it's ugly but if ts doesn't support it let's just add more when we need then.